astropy/units/format/generic.py

```diff
@@ -454,7 +455,22 @@ def _parse_unit(cls, s, detailed_exception=True):
         registry = core.get_current_unit_registry().registry
         if s == '%':
             return registry['percent']
-        elif s in registry:
+
+        if not s.isascii():
+            # common normalization of unicode strings to avoid
+            # having to deal with multiple representations of
+            # the same character. This normalizes to "composed" form
+            # and will e.g. convert OHM SIGN to GREEK CAPITAL LETTER OMEGA
+            s = unicodedata.normalize('NFC', s)
```

Comment:
Do we still need that here, given that it is done in `parse` already? I'd move the comment there too.